hazelcast/src/test/java/com/hazelcast/client/multimap/ClientMultiMapTest.java

```diff
@@ -65,6 +78,137 @@ public void testPut() {
         assertTrue(mm.put(key, 1));
     }
 
+    protected <K, V> EntryListener<K, V> putAllEntryListenerBuilder(Consumer<EntryEvent<K, V>> f) {
+        return new EntryAdapter<K, V>() {
+            public void entryAdded(EntryEvent<K, V> event) {
+                f.accept(event);
+            }
+
+            public void entryRemoved(EntryEvent<K, V> event) {
+            }
+
+            public void entryEvicted(EntryEvent<K, V> event) {
+                entryRemoved(event);
+            }
+
+            @Override
+            public void mapEvicted(MapEvent event) {
+            }
+
+            @Override
+            public void mapCleared(MapEvent event) {
+            }
+        };
+    }
+
+    protected void testMultiMapPutAllSetup() {
+        MultiMapConfig multiMapConfig1 = new MultiMapConfig()
+                .setName("testMultiMapList")
+                .setValueCollectionType(MultiMapConfig.ValueCollectionType.LIST)
+                .setBinary(false);
+        MultiMapConfig multiMapConfig2 = new MultiMapConfig()
+                .setName("testMultiMapSet")
+                .setValueCollectionType(MultiMapConfig.ValueCollectionType.SET)
+                .setBinary(false);
+        MultiMapConfig multiMapConfig3 = new MultiMapConfig()
+                .setName("expectedMultiMap1")
+                .setValueCollectionType(MultiMapConfig.ValueCollectionType.LIST)
+                .setBinary(false);
+        client.getConfig()
+                .addMultiMapConfig(multiMapConfig1)
+                .addMultiMapConfig(multiMapConfig2)
+                .addMultiMapConfig(multiMapConfig3);
+    }
+
+    public void testMultiMapPutAllTemplate(Map<String, Collection<? extends Integer>> expectedMultiMap,
+                                           Consumer<MultiMap<String, Integer>> putAllOperation1) throws InterruptedException {
```

Comment:
Some naming suggestions to make the code a little bit more easy to understand 
Please remove the numbers from the names where it is not necessary . (When there is a number, I automatically search for the second one. )
like `putAllOperation1` -> `putAllOperation`